### ISSUE TITLE
[9.0.0] ReferenceError: module is not defined 

### ISSUE THREAD

**Issue by kulakowka** [jaydenseric/apollo-upload-client#118]
<img width="586" alt="screen shot 2018-09-27 at 1 15 33 pm" src="https://user-images.githubusercontent.com/557190/46139531-770bb780-c257-11e8-9e23-229feaeb2866.png">

I tried to update and got an error.

```json
"react-scripts": "^2.0.0",
"apollo-upload-client": "^9.0.0",
```

**Comment by jaydenseric** [comment]
This is a CRA bug, not an issue with `apollo-upload-client`. They need to support the same file extensions Node.js and Webpack does. `.mjs` should be resolved before `.js`. This is not about supporting `.mjs` in _project_ files, but allowing npm dependencies to bundle correctly with Webpack.

https://github.com/facebook/create-react-app/blob/3ae3cf3678725f86fe34234e59088a65a9458b8b/packages/react-scripts/config/webpack.config.prod.js#L200

Here is an example of what it should look like: https://github.com/zeit/next.js/blob/7.0.1-canary.0/build/webpack.js#L155

This is a similar issue to https://github.com/jaydenseric/apollo-upload-client/issues/117.

Don't worry, I'm sure they will fix it because their flawed config will result in bundling errors for other dependencies too, such as `graphql` - a popular package maintained by Facebook.

**Comment by dirkdevriendt** [comment]
FYI, the import works now, but the library still runs into an issue when trying to access functionality from `extract-files`:
```
TypeError: extractFiles is not a function
    at ApolloLink.request (index.js:54)
    at ApolloLink.request (link.js:33)
    at link.js:55
...
```
Which points to line 54 in `node_modules/apollo-upload-client/lib/index.js`:
```js
    var files = extractFiles(body);
```
I assume after failing to fullfill the require in line 16:
```js
var _require3 = require('extract-files'),
    extractFiles = _require3.extractFiles,
    ReactNativeFile = _require3.ReactNativeFile;
```

I'm guessing this is still an incompatibility between the latest CRA tooling and the require / import statements or is something else happening here?
Side question: am I understanding the documentation correctly that the link provided by `createUploadLink` should replace the one given by `createHttpLink`, with the same endpoint?

Final note: I am also using `graphql (@14.0.1)` in this same project and it did keep on working through these upgrades. Which does't prove anything, but I thought you might want to know.

Thanks again!

**Comment by jaydenseric** [comment]
Config still looks wrong to me: https://github.com/facebook/create-react-app/blob/react-scripts%402.0.1/packages/react-scripts/config/webpack.config.prod.js#L200

**Comment by karensg** [comment]
Here is the discussion about the final version of react-scripts:
https://github.com/facebook/create-react-app/issues/5103
I believe that they require dependencies to build themselves.

**Comment by jaydenseric** [comment]
Not sure what you mean, the dependency is precompiled: https://registry.npmjs.org/extract-files/-/extract-files-4.0.0.tgz

**Comment by karensg** [comment]
Having the same issue as @dirkdevriendt. 
In that CRA issue, they say that .mjs support is dropped.

**Comment by dirkdevriendt** [comment]
Indeed, I think, as @jaydenseric indicates, it's really about the .mjs extension. I "fixed" it by renaming all *.mjs files to *.js in my local version of the repo.

:-) This may again have caused other issues down the line, but where would the fun be if that didn't happen...

**Comment by dirkdevriendt** [comment]
FYI: my current understanding is that CRA is waiting on Jest support which is waiting on esm support...
ref:
https://github.com/facebook/create-react-app/issues/4317#issuecomment-424020579
https://github.com/facebook/jest/issues/4842#issuecomment-401877985
https://github.com/facebook/jest/issues/7018#issuecomment-425466787

**Comment by jaydenseric** [comment]
Next.js does not support `.mjs` for project files, but it ***must***, and does, support it for `node_modules` dependencies. CRA has apparently thrown the baby out with the batch water, they should add `.mjs` to the list of resolve extensions (before `.js`).

**Comment by karensg** [comment]
They fixed it in 2.0.4
https://github.com/facebook/create-react-app/blob/master/CHANGELOG.md
